### ISSUE TITLE
Unable to connect to Switchbot S10 vacuum

### ISSUE THREAD

**Issue by harryoffer** [AdyRock/com.switchbot#111]
Just received my S10 vacuum.
Able to add it to the Switchbot IOS and Android apps and use it from there.
Unable to add to Homey Pro 2023

I have v1.1.51 of the Homey SwitchBot app

If I click configure and "Get Hub Devices", I can see my device ID
{
  "deviceList": [
    {
      "deviceId": "B0E9FE001ECB",
      "deviceName": "s10",
      "enableCloudService": true,
      "hubDeviceId": ""
    }
  ],
  "infraredRemoteList": []
}

Going to Get status and putting in the above ID, I get this error
Error: No OAuth2 Client Found

I was prompted to put in my swichbot credentials which I did and accepted, but got a screenshot "no devices found"

I have tried restarting both the homey switchbot app and the actual homey.
Still seeing the same issue

**Comment by AdyRock** [comment]
Unfortunately, there is no deviceType field, so Homey has no idea what that device is.
Can you check if there are any updates from SwitchBot for the vacuum?

**Comment by harryoffer** [comment]
I’ve got the latest firmware


settle still there


Seems crazy as the image shown when connecting a new robot vacuum is
definitely the S10 as it’s their first vacuum with a wet and dry station
(shown on the image)

**Comment by AdyRock** [comment]
I have requested support from Switchbot but they are not fast to respond.

**Comment by harryoffer** [comment]
Not a prob
Thanks, I appreciate it

**Comment by jonasvekeman** [comment]
Hi

Did you receive any response from Switchbot regarding the [S10](https://eu.switch-bot.com/pages/switchbot-floor-cleaning-robot-s10) vacuum?

Let me known if we can assist in any way to get this device integrated with homey.

Kind regards

**Comment by AdyRock** [comment]
Nothing yet. They are not quick when it comes to the external automation support.

**Comment by jonasvekeman** [comment]
Yeah I noticed. I temporary solved it by using the Switchbot scenes in their app. I created a scene that starts the S10 bot and start that specific scene in the Homey flows. It's not ideal to get proper feedback from the bot but not much more we can do atm :)

**Comment by heikobischoff** [comment]
This is my output for "Get Devices":

{
  "deviceList": [
    {
      "deviceId": "<removed>",
      "deviceName": "Vacuum",
      "deviceType": "Robot Vacuum Cleaner S10",
      "enableCloudService": true,
      "hubDeviceId": ""
    },
    {
      "deviceId": "<removed>",
      "deviceName": "SwitchBotHub 2",
      "deviceType": "Hub 2",
      "enableCloudService": true,
      "hubDeviceId": "000000000000"
    }
  ],
  "infraredRemoteList": [],
  "sceneList": []
}

And this is the output when I run "Get Status" for the Device ID of the S10:

{ "deviceId": "<removed>", "deviceType": "WoSweeperOrigin", "workingStatus": "ChargeDone", "onlineStatus": "online", "battery": 100, "waterBaseBattery": 100, "taskType": "standBy" }

It seems the information is there, but I'm unable to add the device to homey. The error message is that no new devices are found.

**Comment by AdyRock** [comment]
The API details for the S10 have recently been released, so I will add support for it soon.

**Comment by D117D** [comment]
Is it possible to connect the S10 yet?

**Comment by harryoffer** [comment]
Will check this weekend and get back to you

sent while mobile.

**Comment by AdyRock** [comment]
It should be available in the test version.

**Comment by jonasvekeman** [comment]
Thanks a lot to add support!

I was able to add my device, need a bit more testing but seems to be working fine so far.

What I would love as addition is the ability to set 
- The cleaning mode: I believe in the API it's the startClean function (sweep or sweep_mop). 
- The mop waterlevel: API changeParam function (waterLevel, 1 = moist, 2 = wet). This variable is also used in the startClean function. 

Maybe a minor thing: small typo in "Power Level Quite" -> "Power Level Quiet"

**Comment by heikobischoff** [comment]
While testing today, I noticed a few issues:

* The "Start" command doesn’t seem to work. The debug window shows that the command was sent successfully, but the robot doesn’t begin cleaning.
* The "Stop" command also isn’t functioning for me while the robot is cleaning. The only way I can stop the robot is by using the "Dock" command.
* I wanted to create a flow with a condition based on the robot's state. When the robot is running, its state is displayed as "Clearing," but in an advanced flow's state condition, only the option "Cleaning" is available. This seems like a potential typo or mismatch.

Thanks for looking into these!
